src/SQLite.Net/TableMapping.cs

```diff
@@ -50,11 +55,39 @@ public TableMapping(Type type, IEnumerable<PropertyInfo> properties, CreateFlags
             var cols = new List<Column>();
             foreach (var p in props)
             {
-				var ignore = infoProvider.IsIgnored (p);
+                var ignore = infoProvider.IsIgnored(p);
+                if (ignore) continue;
 
-                if (p.CanWrite && !ignore)
-                    cols.Add(new Column(p, createFlags));
+                var memberType = infoProvider.GetMemberType(p);
+                // Check if this is a ValueTuple<...>
+                if (memberType.GetInterface(nameof(ITuple)) != null && memberType.IsValueType)
+                {
+                    // If it is, create a column per member of the value tuple.
+                    var names = p.GetCustomAttribute<TupleElementNamesAttribute>();
+                    var args = memberType.GetGenericArguments();
+                    for (var i = 0; i < args.Length; ++i)
+                    {
+                        cols.Add(new Column(
+                            p,
+                            createFlags,
+                            i,
+                            args[i],
+                            names?.TransformNames[i] ?? $"Item{i + 1}",
+                            infoProvider));
+                    }
+                }
+                else
+                {
+                    cols.Add(new Column(
+                        p,
+                        createFlags,
+                        -1,
+                        null,
+                        null,
+                        infoProvider));
```

Comment:
why not keep this old constructor for this case? it's kind of clunky to read now

Reply:
Because we were not passing the infoProvider into the columns, which is odd and a bug.